qcodes/dataset/sqlite_base.py

```diff
@@ -1904,6 +1911,39 @@ def get_metadata(conn: SomeConnection, tag: str, table_name: str):
                             "result_table_name", table_name)
 
 
+def get_metadata_from_run_id(conn: SomeConnection, run_id: int) -> Dict:
+    """
+    Get all metadata associated with the specified run
+    """
+    # TODO: promote snapshot to be present at creation time
```

Comment:
does this TODO mean upgrading the schema to have snapshot column in it?

Reply:
Yes, exactly. I think it's just a blunder that it is not already there.